embassy-imxrt/src/gpio.rs

```diff
@@ -0,0 +1,1058 @@
+    fn from(val: bool) -> Self {
+        match val {
+            true => Self::High,
+            false => Self::Low,
+        }
+    }
+}
+
+impl From<Level> for bool {
+    fn from(level: Level) -> bool {
+        match level {
+            Level::Low => false,
+            Level::High => true,
+        }
+    }
+}
+
+/// Interrupt trigger levels.
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum InterruptType {
+    /// Trigger on level.
+    Level,
+    /// Trigger on edge.
+    Edge,
+}
+
+#[cfg(feature = "rt")]
+#[interrupt]
+#[allow(non_snake_case)]
+fn GPIO_INTA() {
+    irq_handler(&GPIO_WAKERS);
+}
+
+#[cfg(feature = "rt")]
+struct BitIter(u32);
+
+#[cfg(feature = "rt")]
+impl Iterator for BitIter {
+    type Item = u32;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.0.trailing_zeros() {
+            32 => None,
+            b => {
+                self.0 &= !(1 << b);
+                Some(b)
+            }
+        }
+    }
+}
+
+#[cfg(feature = "rt")]
+fn irq_handler(port_wakers: &[Option<&PortWaker>]) {
+    let reg = unsafe { crate::pac::Gpio::steal() };
+
+    for (port, port_waker) in port_wakers.iter().enumerate() {
+        let Some(port_waker) = port_waker else {
+            continue;
+        };
+
+        let stat = reg.intstata(port).read().bits();
+        for pin in BitIter(stat) {
+            // Clear the interrupt from this pin
+            reg.intstata(port).write(|w| unsafe { w.status().bits(1 << pin) });
+            // Disable interrupt from this pin
+            reg.intena(port)
+                .modify(|r, w| unsafe { w.int_en().bits(r.int_en().bits() & !(1 << pin)) });
+
+            let Some(waker) = port_waker.get_waker(pin as usize) else {
+                continue;
+            };
+
+            waker.wake();
+        }
+    }
+}
+
+/// Initialization Logic
+/// Note: GPIO port clocks are initialized in the clocks module.
+pub(crate) fn init() {
+    // Enable GPIO clocks
+    enable_and_reset::<peripherals::HSGPIO0>();
+    enable_and_reset::<peripherals::HSGPIO1>();
+    enable_and_reset::<peripherals::HSGPIO2>();
+    enable_and_reset::<peripherals::HSGPIO3>();
+    enable_and_reset::<peripherals::HSGPIO4>();
+    enable_and_reset::<peripherals::HSGPIO5>();
+    enable_and_reset::<peripherals::HSGPIO6>();
+    enable_and_reset::<peripherals::HSGPIO7>();
+
+    // Enable INTA
+    interrupt::GPIO_INTA.unpend();
+
+    // SAFETY:
+    //
+    // At this point, all GPIO interrupts are masked. No interrupts
+    // will trigger until a pin is configured as Input, which can only
+    // happen after initialization of the HAL
+    unsafe { interrupt::GPIO_INTA.enable() };
+}
+
+/// Input Sense mode.
+pub trait Sense: Sealed {}
+
+/// Sense Enabled Flex pin.
+///
+/// This is a true flex pin as the input buffer is enabled.
+/// It can sense its own level when even when configured as an output pin.
+pub struct SenseEnabled;
```

Comment:
```suggestion
pub enum SenseEnabled {}
```

So that this type can't be constructed.

Reply:
Fixed